animareflection/ui
# Foundation

## Description

##### Task link: https://trello.com/c/qwSmnpZQ/68-ui-kit

- Added initial project boilerplate, of primary interest:
  - tsup for bundling
  - Panda CSS for framework-agnostic styling + Ark for framework-agnostic headless components
  - Storybook for development and documentation of design system and UI components
- Added Vercel Storybook preview URLs *only to preview envs* (e.g. PRs, no production deployment on `master`)
- Added CI
  - Unit test runner
  - Coverage report runner with PR reporting integration

**⚠️ Important notes:**
- This PR does not implement release cycle management, publishing, or anything of that nature. That will soon be implemented in a future PR
- The design system "patterns" are simply set to get a base reference started (e.g. the button component recipe code) and are subject to change as we collectively see fit. In other words, look at the content of this PR purely as a placeholder rather than "CSS styles and other patterns set in stone"
- Although the current library export is only a button, Panda utility patterns (e.g. layout patterns) will be exported in the future
- `panda codegen` is intentionally not added to the `prepare` script and is instead prepended to necessary scripts due to constant cycling when running `yarn dev`

## Test Steps

- Locally clone this repo and switch to this branch, then follow the `Local Development` steps in `README.md` and make sure the steps work properly
- Verify unit tests work locally (`yarn test`)
- Verify the Storybook Vercel preview deployment properly renders the ANIREF design system Storybook
